libvast/builtins/formats/json.cpp

```diff
@@ -709,62 +711,85 @@ class json_parser final : public plugin_parser {
+                    ctrl, not args_.unnest_separator.empty());
     auto schema = std::optional<type>{};
-    if (cfg_.schema) {
+    if (args_.schema) {
       const auto found
         = std::find_if(schemas.begin(), schemas.end(), [&](const type& schema) {
             for (const auto& name : schema.names()) {
-              if (name == cfg_.schema->inner) {
+              if (name == args_.schema->inner) {
                 return true;
               }
             }
             return false;
           });
       if (found == schemas.end()) {
-        diagnostic::error("failed to find schema `{}`", cfg_.schema->inner)
-          .primary(cfg_.schema->source)
+        diagnostic::error("failed to find schema `{}`", args_.schema->inner)
+          .primary(args_.schema->source)
           // TODO: Refer to the show operator once we have that.
           .note("use `tenzir-ctl show schemas` to show all available schemas")
           .emit(ctrl.diagnostics());
         return {};
       }
       schema = *found;
     }
-    if (cfg_.use_ndjson_mode) {
+    if (args_.use_ndjson_mode) {
       return make_parser(to_padded_lines(std::move(input)),
-                         cfg_.unnest_separator, cfg_.selector.has_value(),
-                         schema, not cfg_.no_infer,
+                         args_.unnest_separator, args_.selector.has_value(),
+                         schema, not args_.no_infer,
                          ndjson_parser<FieldValidator>{
                            ctrl,
-                           cfg_.selector,
+                           args_.selector,
                            schema,
                            std::move(schemas),
                            std::move(field_validator),
-                           not cfg_.no_infer,
+                           not args_.no_infer,
                          });
     }
-    return make_parser(std::move(input), cfg_.unnest_separator,
-                       cfg_.selector.has_value(), schema, not cfg_.no_infer,
+    return make_parser(std::move(input), args_.unnest_separator,
+                       args_.selector.has_value(), schema, not args_.no_infer,
                        default_parser<FieldValidator>{
                          ctrl,
-                         cfg_.selector,
+                         args_.selector,
                          schema,
                          std::move(schemas),
                          std::move(field_validator),
-                         not cfg_.no_infer,
+                         not args_.no_infer,
                        });
   }
 
-  config cfg_;
+  parser_args args_;
+};
+
+struct printer_args {
+  std::optional<location> compact_output;
+  std::optional<location> color_output;
+  std::optional<location> monochrome_output;
+  std::optional<location> omit_empty;
+  std::optional<location> omit_nulls;
+  std::optional<location> omit_empty_objects;
+  std::optional<location> omit_empty_lists;
```

Comment:
Just to let you know: For those where you are not using the location, you can just use `bool` here (the argument parser works with both). But it also works as-is. Your choice.

Reply:
I don't have too much of a strong opinion on this. Probably not worth changing now.